### DISCUSSION TITLE
Phase Coherence

### DISCUSSION THREAD

**Discussion by nicko-ai** [magenta/ddsp#300] [Q&A] [answered]
I see you added the StreamingF0Ld class to the ddsp/training/inference.py module.
Can you tell me, how you can use your model in real-time and generate frames with consistent phases and no border effects?

**Comment by jesseengel** [comment, marked as answer]
Hi,

That class is for the upcoming JUCE plugin that does real-time streaming.
The additive synthesis procedure maintains phase coherence by design by
smoothly incrementing the phase between each frame and then just taking the
sin.
